packages/flutter/test/widgets/shortcuts_test.dart

```diff
@@ -1419,4 +1307,526 @@ void main() {
       await tester.sendKeyUpEvent(LogicalKeyboardKey.keyB);
     });
   });
+
+  group('ShortcutRegistrar', () {
+    testWidgets('trigger ShortcutRegistrar on key events', (WidgetTester tester) async {
+      int invokedA = 0;
+      int invokedB = 0;
+      await tester.pumpWidget(
+        ShortcutRegistrar(
+          child: TestCallbackRegistration(
+            shortcuts: <ShortcutActivator, Intent>{
+              const SingleActivator(LogicalKeyboardKey.keyA): _VoidCallbackIntent(() {
+                invokedA += 1;
+              }),
+              const SingleActivator(LogicalKeyboardKey.keyB): _VoidCallbackIntent(() {
+                invokedB += 1;
+              }),
+            },
+            child: Actions(
+              actions: <Type, Action<Intent>>{
+                _VoidCallbackIntent: _VoidCallbackAction(),
+              },
+              child: const Focus(
+                autofocus: true,
+                child: Placeholder(),
+              ),
+            ),
+          ),
+        ),
+      );
+      await tester.pump();
```

Comment:
I assume this pump is needed because of the postframecallback logic to actually activate the shortcuts? If we have to stay with that behavior, we should document on the addAll method that shortcuts will only become active in the next frame (and maybe even tell people that it requires an extra pump in tests?).

It makes me wonder even stronger if we can avoid the frame delay. People will get tripped up by this in testing...